model/state_model.jl

```diff
@@ -11,418 +10,434 @@ using JuMP
-@variable(cge,RA[r in regions]>=sv,start=blueNOTE[:c0][(r,)]) # Representative agent
+@variable(cge,RA[r in set[:r]]>=lo,start = value(c0[r])) ;
+
 
 ###############################
 # -- PLACEHOLDER VARIABLES --
 ###############################
 
-@NLexpression(cge,CVA[r in regions,s in sectors],
-  PL[r]^alpha_kl[r,s] * PK[r,s] ^(1-alpha_kl[r,s]) );
-
-@NLexpression(cge,AL[r in regions, s in sectors],
-  blueNOTE[:ld0][r,s] * CVA[r,s] / PL[r] );
-
-@NLexpression(cge,AK[r in regions,s in sectors],
-  blueNOTE[:kd0][r,s] * CVA[r,s] / PK[r,s] );
-
-  ###
+#cobb-douglas function for value added (VA)
+@NLexpression(cge,CVA[r in set[:r],s in set[:s]],
+  PL[r]^alpha_kl[r,s] * (haskey(PK.lookup[1], (r,s)) ? PK[(r,s)] : 1.0) ^ (1-alpha_kl[r,s]) );
 
-@NLexpression(cge,RX[r in regions,g in goods],
-  (alpha_x[r,g]*PFX^5+alpha_n[r,g]*PN[g]^5+alpha_d[r,g]*PD[r,g]^5)^(1/5) );
+#demand for labor in VA
+@NLexpression(cge,AL[r in set[:r], s in set[:s]], ld0[r,s] * CVA[r,s] / PL[r] );
 
-@NLexpression(cge,AX[r in regions,g in goods],
-  (blueNOTE[:x0][r,g] - blueNOTE[:rx0][r,g])*(PFX/RX[r,g])^4 );
+#demand for capital in VA
+@NLexpression(cge,AK[r in set[:r],s in set[:s]],
+  kd0[r,s] * CVA[r,s] / (haskey(PK.lookup[1], (r,s)) ? PK[(r,s)] : 1.0) );
 
-@NLexpression(cge,AN[r in regions,g in goods],
-  blueNOTE[:xn0][r,g]*(PN[g]/(RX[r,g]))^4 );
+#CES function for output demand - including
+# exports (absent of re-exports) times the price for foreign exchange, 
+# region's supply to national market times the national market price
+# regional supply to local market times domestic price
+@NLexpression(cge,RX[r in set[:r],g in set[:g]],
+  (alpha_x[r,g]*PFX^(1 + et_x[r,g])+alpha_n[r,g]*PN[g]^(1 + et_x[r,g])+alpha_d[r,g]*(haskey(PD.lookup[1], (r,g)) ? PD[(r,g)] : 1.0)^(1 + et_x[r,g]))^(1/(1 + et_x[r,g])) );
 
-@NLexpression(cge,AD[r in regions,g in goods],
-  blueNOTE[:xd0][r,g] * (PD[r,g] / (RX[r,g]))^4 );
+#demand for exports via demand function
+@NLexpression(cge,AX[r in set[:r],g in set[:g]], (x0[r,g] - rx0[r,g])*(PFX/RX[r,g])^et_x[r,g] );
 
-  ###
+#demand for contribution to national market 
+@NLexpression(cge,AN[r in set[:r],g in set[:g]], xn0[r,g]*(PN[g]/(RX[r,g]))^et_x[r,g] );
 
-@NLexpression(cge,CDN[r in regions,g in goods],
-  (theta_n[r,g]*PN[g]^(1-2)+(1-theta_n[r,g])*PD[r,g]^(1-2))^(1/(1-2)) );
+#demand for regionals supply to local market
+@NLexpression(cge,AD[r in set[:r],g in set[:g]],
+  xd0[r,g] * ((haskey(PD.lookup[1], (r,g)) ? PD[(r,g)] : 1.0) / (RX[r,g]))^et_x[r,g] );
 
-#!!!! here replaced first :tm with :tm0
-@NLexpression(cge,CDM[r in regions,g in goods],
-  ((1-theta_m[r,g])*CDN[r,g]^(1-4)+theta_m[r,g]*
-  (PFX*(1+blueNOTE[:tm0][r,g])/(1+blueNOTE[:tm0][r,g]))^(1-4))^(1/(1-4)) 
-  );
+# CES function for tradeoff between national and domestic market
+@NLexpression(cge,CDN[r in set[:r],g in set[:g]],
+  (theta_n[r,g]*PN[g]^(1-es_d[r,g])+(1-theta_n[r,g])*(haskey(PD.lookup[1], (r,g)) ? PD[(r,g)] : 1.0)^(1-es_d[r,g]))^(1/(1-es_d[r,g])) );
```

Comment:
Great to know, thank you. I need to make some adjustments to the VScode settings, I think, so that I can see the lines and ensure indents are correct. I had problems going back and forth between Emacs and VScode, but now working mostly in VScode.